### PR TITLE
Nweinmeister/1564 ua versioning

### PR DESCRIPTION
**Pull request by nweinmeister**
Resolves #1818

### Description
Add business logic for updating with versioning the the unrecognized appellant

### Code Documentation Updates
- [ ] Add or update code comments at the top of the class, module, and/or component.

### CODE REVIEW

File: app/models/unrecognized_appellant.rb [hunk 1]
```diff
@@ -27,8 +27,24 @@ def first_version
     versions.where.not(id: current_version_id).first || self
   end
 
-  def update_with_versioning!(_params)
-    # Will be modified to perform update. _params being used as argument here to pass code climate inspection
-    self
+  def update_with_versioning!(params, user)
+    transaction do
+      create_version
+      update(params.except(:unrecognized_party_detail).merge(created_by: user))
```
Comment (wdrougas-tista) [inline app/models/unrecognized_appellant.rb:33]: TIL you could do this!

File: spec/controllers/unrecognized_appellant_controller_spec.rb [hunk 2]
```diff
@@ -17,6 +23,56 @@
...
+        expect(ua.current_version.created_by).to eq user
+        expect(ua.first_version.created_by).to eq original_created_by
+
+        expect(ua.current_version.unrecognized_party_detail.address_line_1).to eq updated_address_1
+        expect(ua.first_version.unrecognized_party_detail.address_line_1).to eq original_address_line_1
+        expect(ua.versions.count).to eq 2
+      end
+      it "should return the non-updated version with a 400 if one of the updates fails" do
+        # Default the return for the let(:ua)
+        allow_any_instance_of(UnrecognizedAppellant).to receive(:update).and_return(:default)
```
Comment (wdrougas-tista) [inline spec/controllers/unrecognized_appellant_controller_spec.rb:58]: Is :default here `false`?

**Review by wdrougas-tista — APPROVED** [review]
LGTM

File: spec/controllers/unrecognized_appellant_controller_spec.rb (hunk 2 above)
Comment (nweinmeister) [reply spec/controllers/unrecognized_appellant_controller_spec.rb:58]: :default just allows the method to actually be exercised. Kind of annoying that you have to do this but apparently if you specify a `.with` like i did on the next line, you have to specify what happens if the method is called with something besides what you specify. So here I just set let it do what it's supposed to do

File: app/models/unrecognized_appellant.rb (hunk 1 above)
Comment (nweinmeister) [reply app/models/unrecognized_appellant.rb:33]: 🎉

**Pull request merged**